src/eip7594/eip7594.c

```diff
@@ -530,6 +533,73 @@ static C_KZG_RET compute_weighted_sum_of_commitments(
     return ret;
 }
 
+/**
+ * Compute the inverse coset factor h_k^{-1},
+*  where `h_k` is the coset factor for cell with index `k`.
+ *
+ * @param[out]  inv_coset_factor_out  Pointer to store the computed inverse coset factor
+ * @param[in]   cell_index            The index of the cell
+ * @param[in]   s                     The trusted setup
+ */
+static void get_inv_coset_shift_for_cell(
+    fr_t *inv_coset_factor_out,
+    uint64_t cell_index,
+    const KZGSettings *s
+) {
+    /*
+     * Get the cell index in reverse-bit order.
+     * This index points to this cell's coset factor h_k in the roots_of_unity array.
+     */
+    uint64_t cell_idx_rbl = CELL_INDICES_RBL[cell_index];
+
+    /*
+     * Observe that for every element in roots_of_unity, we can find its inverse by
+     * accessing its reflected element.
+     *
+     * For example, consider a multiplicative subgroup with eight elements:
+     * roots = {w^0, w^1, w^2, ... w^7, w^0}
+     * For a root of unity in roots[i], we can find its inverse in roots[-i].
+     */
+    uint64_t N = FIELD_ELEMENTS_PER_EXT_BLOB;
+    uint64_t inv_coset_factor_idx = (N - cell_idx_rbl) % N;
```

Comment:
As discussed in telegram:
```suggestion
    uint64_t inv_coset_factor_idx = FIELD_ELEMENTS_PER_EXT_BLOB - cell_idx_rbl;
```